client/state/purchases/selectors/get-by-purchase-id.js

```diff
@@ -8,5 +9,7 @@ import 'calypso/state/purchases/init';
  * @param   {number} purchaseId  the purchase id
  * @returns {import('calypso/lib/purchases/types').Purchase|undefined} the matching purchase if there is one
  */
-export const getByPurchaseId = ( state, purchaseId ) =>
-	getPurchases( state ).find( ( purchase ) => purchase.id === purchaseId );
+export const getByPurchaseId = createSelector(
+	( state, purchaseId ) => getPurchases( state ).find( ( purchase ) => purchase.id === purchaseId ),
+	[ getPurchases ]
+);
```

Comment:
This looks unnecessary. Can use `getPurchases` (which is already memoized) in the `createSelector` dependencies in `usePricingMetaForGridPlans`